storage/csi-cloning.md

```diff
@@ -0,0 +1,124 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs] 
+
+## Summary
+In the past storage drivers have been provided either in-tree or via external provisioners. With the move to advocating CSI as the way to deliver storage drivers, Red Hat needs to ensure the CSI drivers have comparable functionality to the in-tree drivers.
+
+For the purpose of this enhancement, a Clone is defined as a duplicate of an existing Volume that can be consumed as any standard Volume would be. The only difference is that upon provisioning, rather than creating a "new" empty Volume, the back end device creates an exact duplicate of the specified Volume.
+
+## Motivation
+
+* It has to be assumed that at some point in-tree drivers will be removed from Kubernetes, we need to continue to make the drivers available and CSI is the way to do this.
+* As in-tree drivers move to CSI, we need to ensure that there is no loss in functionality by moving to a CSI driver.
+* Features like Cloning are common in most storage devices, not only is the capability available on most devices, it's also frequently used in various use cases whether it be for duplicating data or to use as a disaster recovery method.
+
+### Goals
+
+* Backport the cloning features in the upstream external-provisioner.
+* Package and ship a downstream image of a sidecar including these features.
+
+### Non-Goals
+
+* This enhancement does NOT propose the addition of other types of DataSource including Populators
+* This enhancement does NOT propose support for special cases like "out of band" cloning (support for back ends that don't have Cloning features), that sort of implementation would fall under Populators.
+* This enhancement does NOT propose any ability to shrink a PVC during a Clone request (e.g. it's considered an invalid request to clone PVC-a with a size of 10Gib to a PVC with a requested size of less than 10Gib, expansion is "ok" if the driver supports it but it's not required)
+* This enhancement does NOT propose adding any ability to transfer a Clone to a different Namespace, the new PVC (Clone) will be in the same Namespace as the origin that was specified. This also means that since this is namespaced, a user can not request a Clone of a PVC that is another Namespace. A user can only request a Clone of PVCs in his or her Namespace.
+* Cloning will only be available in CSI, cloning features will NOT be added to existing in-tree plugins or Flex drivers
+* Cloning will only be available within the same storage class (see [Implementation Details](#implementation-detailsnotesconstraints-optional) section for more info)
+## Proposal
+
+Add the Core object PVC to the allowed types for PVC DataSource.  Currently API validation only allows Snapshot Object Types, this proposal is to also add the Core PersistentVolumeClaim object as an accepted DataSource entry.
```

Comment:
We're not *implementing* the cloning feature here.

IMO, we should propose we release our csi-external-provisioner sidecar image with cloning support (enabled by default) as tech-preview. We provide it to our internal Red Hat teams (CNV, OCS) for their drivers. No other consumer is supported (even though the sidecar can be used by any CSI driver).

We don't need to provide any technical details *how* the API looks like, a reference to upstream KEP or documentation is IMO enough.

Reply:
I've updated the Proposal to include a link to the upstream enhancement and documentation.